src/XMakeBuildEngine/UnitTestsPublicOM/Definition/ProjectItem_Tests.cs

```diff
@@ -450,44 +450,100 @@ public void ExcludeVectorWithIncludeVector()
             ObjectModelHelpers.AssertItems(new[] { "a", "b", "c", "z", "a", "c", "u" }, items);
         }
 
-        /// <summary>
-        /// Include and Exclude containing wildcards
-        /// </summary>
-        [Fact]
-        public void Wildcards()
-        {
-            string directory = null;
-            string file1 = null;
-            string file2 = null;
-            string file3 = null;
-
+        private const string ExcludeWithWildCards = @"
+                    <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003' >
+                        <ItemGroup>
+                            <i Include='{0}' Exclude='{1}'/>
+                        </ItemGroup>
+                    </Project>
+                ";
+        [Theory]
+        [InlineData(
+            ExcludeWithWildCards,
+            "a.*",
+            "*.1",
+            new[] { "a.1", "a.2", "a.1" },
+            new[] { "a.2" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*.cs",
+            @"a\**",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs" },
+            new[] { "1.cs" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"**\*",
+            @"**\b\**",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs" },
+            new[] { "1.cs", @"a\2.cs", "build.proj" })]
+        [InlineData(ExcludeWithWildCards,
+            @"**\*",
+            @"**\b\**\*.cs",
+            new[] { "1.cs", @"a\2.cs", @"a\b\3.cs", @"a\b\c\4.cs", @"a\b\c\5.txt" },
+            new[] { "1.cs", @"a\2.cs", @"a\b\c\5.txt", "build.proj" })]
+        [InlineData(
+            ExcludeWithWildCards,
+            @"src\**\proj\**\*.cs",
+            @"src\**\proj\**\none\**\*",
+            new[]
+            {
+                "1.cs",
+                @"src\2.cs",
```

Comment:
Yes. Next step is to change that test so it runs the same data with slashes and backslashes (and maybe a random combination of both in the same path?)